llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -563,3 +602,16 @@ AIEHazardRecognizer::getSelectedAltOpcode(MachineInstr *MI) const {
     return It->second;
   return std::nullopt;
 }
+
+MemoryBankBits AIEHazardRecognizer::getMemoryBanks(MachineInstr *MI) const {
+  unsigned MemoryBankUsed = 0;
+  if (MI->mayLoad() || MI->mayStore()) {
+    const AIEBaseSubtarget &STI = AIEBaseSubtarget::get(*MI->getMF());
+    for (auto &MMO : MI->memoperands()) {
+      unsigned MemoryBank =
+          (STI.getMemoryBanksFromAddressSpace(MMO->getAddrSpace())).to_ulong();
+      MemoryBankUsed |= MemoryBank;
+    }
+  }
+  return MemoryBankUsed;
```

Comment:
Nit To limit indentation, I'd prefer an early return like
```
if (!MI->mayLoad())
  return {};
```

I also think we shouldn't care about stores, as mentioned in https://github.com/Xilinx/llvm-aie/pull/95/files#r1668455116